### ISSUE TITLE
[BUG] Sending audio to two other peers is distorted on receiver's end

### ISSUE THREAD

**Issue by streunerlein** [Unity-Technologies/com.unity.webrtc#545]
**Describe the bug**
When sending ~~microphone input~~ _any audio_ (see my answer [below](https://github.com/Unity-Technologies/com.unity.webrtc/issues/545#issuecomment-947007716)) two other peers, the output on the third peer is highly distorted as if the sample rate was wrong. This is reproducible >50% of the time and is the case for local signaling as well as networked signaling with network delays. 

**To Reproduce**
1. Create a new Unity Project and import UnityWebRTC `2.4.0-exp4` with Examples
2. Open the `MultiplePeerConnections` scene in `Assets/Samples/WebRTC/2.4.0-exp4/Example/MultiplePeerConnections`
3. Replace the sample audio clip with microphone input as follows.

**Update:** You can also just use this clip, which is a microphone recording from inside Unity: https://dominiquesandoz.ch/share/test.wav

In *MultiplePeerConnectionsSample.cs*, replace the `Setup()` function:

```
// ...
        private void Setup()
        {
            Debug.Log("Set up source/receive streams");
            sourceStream = new MediaStream();

            var videoTrack = cam.CaptureStreamTrack(streamingSize.x, streamingSize.y, 0);
            sourceStream.AddTrack(videoTrack);
            sourceImage.texture = cam.targetTexture;

            sourceAudio.clip = audioclipStereoSample;
            sourceAudio.loop = true;
            sourceAudio.Play();
            var audioTrack = new AudioStreamTrack(sourceAudio);
            sourceStream.AddTrack(audioTrack);

            startButton.interactable = false;
            callButton.interactable = true;
        }
// ...
```

with

```
// ...
        private void Setup()
        {
            Debug.Log("Set up source/receive streams");
            sourceStream = new MediaStream();

            var videoTrack = cam.CaptureStreamTrack(streamingSize.x, streamingSize.y, 0);
            sourceStream.AddTrack(videoTrack);
            sourceImage.texture = cam.targetTexture;

            var localAudio = Microphone.Start("", true, 1, 48000);
    
            while (!(Microphone.GetPosition("") > 0)) {}

            sourceAudio.clip = localAudio;
            sourceAudio.loop = true;
            sourceAudio.Play();
            var audioTrack = new AudioStreamTrack(sourceAudio);
            sourceStream.AddTrack(audioTrack);

            startButton.interactable = false;
            callButton.interactable = true;
        }
// ...
```
4. Run the scene and press "Start", then "Call"
5. You will hear your voice/microphone input 3 times: the local peer, the first remote peer and the second remote peer.
6. One of the playbacks is extremely distorted (probably the third under `Canvas/grid layout/ReceiveRawImage2`)

Mute AudioSources via the hierarchy/inspector to hear them isolated.
You may have to introduce some [load](https://github.com/Unity-Technologies/com.unity.webrtc/issues/545#issuecomment-947355613) on the scene to have the effect, it is harder to reproduce on very high framerates.

**Expected behavior**
All 3 audio playbacks should be the undistorted microphone inputs.

**Video**
I have summarized this in a video (watch with sound). It's basically me saying _Hello_ repeatedly. You'll hear two clean versions of _Hello_ stemming from the local microphone playback audiosource and one peer and a deep, growling version stemming from the second remote peer; by muting the audiosources one after the other it is clear that it is one of the peers outputting a deep, stretched version. The echos in the video can be ignored, the video was recorded with audio going to the loudspeakers. The issue stays the same though also when using headphones.

[![Preview](https://img.youtube.com/vi/pvmGQhH4L10/0.jpg)](https://youtu.be/pvmGQhH4L10)
(Youtube: https://youtu.be/pvmGQhH4L10)

**Environment:**
 - OS: Windows, Mac
 - Unity Version 2021.2b15, 2020.3.20f1
 - Package version 2.4.0-exp4

**Comment by karasusan** [comment]
@streunerlein @bastienstefani
Thanks for your sharing.

Sounds like this is a platform specified issue... 
How about it with Unity2020?

**Comment by streunerlein** [comment]
I also have the feeling it is connected to framerate. This morning when I've turned on my PC and had nothing else running I could almost not reproduce the issue. Then I've added some more "load" to the scene (see below) to bring FPS down to ~100 I've had a perfect reproduction of the bug.

The load:
```
        private void Update()
        {
// ...
            for (int i = 0; i < 100; i++)
            {
                Debug.Log(i);
            }
        }
```

**Comment by streunerlein** [comment]
@karasusan Can I supply more data on this or is it too specific to be a concern to other users of Unity-Webrtc?

**Comment by Pacman18** [comment]
plz this issue modify file commit, i want test to this

**Comment by karasusan** [comment]
@Pacman18 
We will investigate the issue next week.

**Comment by streunerlein** [comment]
So excited for 2.4.0-exp.5 💯💯💯!!!

**Comment by karasusan** [comment]
@streunerlein
We released the new version 2.4.0-exp.5.
Please check it out and if your issue is not solved, please notify me.
